### DISCUSSION TITLE
problem with number of input channels, using spleen_segmentation_3d.ipynb with own dataset

### DISCUSSION THREAD

**Discussion by esther1262** [Project-MONAI/tutorials#1443] [Q&A] [answered]
Hi, I try to train the model using the tutorial _spleen_segmentation_3d.ipynb_ with my own mri dataset but I keep having errors. I wonder if anyone can help. I saved my images in nii.gz. Each has dimension 256x256x16. Labelmap (binary) is also saved in nii.gz as labelmap volume rather than scalar volume. The labelmap shows 1 for tumor and 0 otherwise. Dimension of masked labelmap is the same as image. 
I loaded both image and label using `nib.load` and checked the shapes, they are all 256x256x16. That means there's only one channel correct?

I used the same augmentation transformation for both training and validation sets, and modified some parameters:
```
augmentation_transforms = Compose(
    [
        LoadImaged(keys=["image", "label"]),
        EnsureTyped(keys=["image", "label"]),
        RandFlipd(keys=["image", "label"], prob=0.2, spatial_axis=0),
        RandFlipd(keys=["image", "label"], prob=0.2, spatial_axis=1),
        RandFlipd(keys=["image", "label"], prob=0.2, spatial_axis=2),
        RandScaleIntensityd(keys="image", factors=0.1, prob=0.5),
        RandShiftIntensityd(keys="image", offsets=0.1, prob=0.5),
        RandGaussianNoised(keys="image", mean=0, std=0.1, prob=0.1),
        RandBiasFieldd(keys="image", coeff_range=(0.2,0.3), prob=0.1)
    ]
)
```
and 

```
UNet_meatdata = {
    "spatial_dims": 3,
    "in_channels": 1,
    "out_channels": 1,
    "channels": (16, 32, 64, 128, 256), 
    "strides": (2, 2, 2, 2),
    "norm": Norm.BATCH
}
```
The remaining codes are pretty much the same as tutorial. 

I got the RuntimeError that _Given groups=1, weight of size [16, 1, 3, 3, 3], expected input[1, 2, 256, 256, 16] to have 1 channels, but got 2 channels instead_.

I googled about this error but most of the solutions are to manually change the parameters in convolutional layers. In this tutorial, we use `UNet_meatdata` and define the model using `UNet(**UNet_meatdata).to(device)`. 
How to fix the issue then?

To check the transforms in dataloader, the tutorial uses ```image, label = (check_data["image"][0][0], check_data["label"][0][0]) ``` and the output is 96x96x96. 
I used ```image, label = (check_data["image"][0], check_data["label"][0]) ``` and outputed torch.size 256x256x16.

**Comment by KumoLiu** [comment, marked as answer]
Hi @esther1262, could you please try `LoadImaged(keys=["image", "label"], ensure_channel_first=True)` in your code and see what comes out? Seems you didn't add a channel dim in the transform chain.
Hope it can help you.
Thanks!

**Reply by esther1262** [reply]
Thanks for your kind help! The codes work now! The tutorial uses ```LoadImaged(keys=["image", "label"]), EnsureChannelFirstd(keys="image"),...```. I thought ensure channel first is optional.

**Reply by KumoLiu** [reply]
Yes, you can use `EnsureChannelFirstd` or `ensure_channel_first=True` in `LoadImaged`. Just remember to add a channel dim.
Thanks!
